client/components/App.jsx

```diff
@@ -111,82 +46,62 @@ class App extends React.Component {
         </li>
       );
     });
-  }
-
-  render() {
-    const {
-      selectedCategory,
-      categories,
-      filters,
-      items,
-      orders,
-      isAllFilters,
-      isOrdersVisible,
-      discounts,
-    } = this.state;
-
-    const categoriesList = this.categoriesList(selectedCategory);
-    const categoryTitle = categories.find((elem) => elem._id === selectedCategory)?.title;
-
-    const activeFilters = filters.filter((elem) => elem.isActive)?.map((elem) => elem._id);
-
-    const filteredItems = activeFilters.length > 0
-      ? items?.filter((elem) => {
-        const intersection = elem.filterIds.filter((x) => activeFilters.includes(x));
-        return (
-          intersection.length > 0
-              && intersection.length <= elem.filterIds.length
-              && intersection.length === activeFilters.length
-        );
-      })
-      : items;
-
-    return (
-      <div className="app">
-        <nav className="side-nav">
-          <h1>P.</h1>
-          <p>categories</p>
-          <ul>{categoriesList}</ul>
-        </nav>
-        <div className="main">
-          <header>
-            <Filters
-              tags={filters}
-              onSwitch={this.switchFilterCallback}
-              onSwitchAll={this.switchDisplayAllCallback}
-              all={isAllFilters}
-            />
-            <OrderStatus orders={orders} onClick={this.switchOrdersCallback} />
-          </header>
-          <List items={filteredItems} title={categoryTitle} onAdd={App.addOrder} />
-        </div>
-        {isOrdersVisible ? (
-          <Orders orders={orders} discounts={discounts} onClose={this.switchOrdersCallback} />
-        ) : (
-          ''
-        )}
+  };
+
+  const getSwitchFilterCallback = () => (id) => dispatch(switchFilter(id));
+  const getSwitchAllCallback = () => () => dispatch(switchAll());
+  const getSwitchOrdersCallback = () => () => dispatch(switchOrders());
+  const getPostOrdersCallback = () => (id) => dispatch(postOrder(id));
+
+  // render
+  const categoriesLst = categoriesList(data.selectedCategory);
+  const categoryTitle = data.categories.find((elem) => elem._id === data.selectedCategory)?.title;
+  const filters = data.filters.map((elem) => ({
+    isActive: activeFilters.includes(elem._id),
+    ...elem,
+  }));
+
+  const filteredItems = activeFilters.length > 0
```

Comment:
@IgorChaikin Could you configure `eslint` to use same code formatting settings, as `prettier`:

<img width="679" alt="image" src="https://user-images.githubusercontent.com/12561302/128700736-b2a8e33f-e42d-4780-9b99-ce1680f7666f.png">

You can try to use `plugin:prettier/recommended` as last item of `extends` array of `eslint` settings